tools/dotnet-msidentity/Program.cs

```diff
@@ -286,6 +312,23 @@ private static async Task<int> HandleClientSecrets(ProvisioningToolOptions provi
                 IsRequired = false
             };
 
+        private static Option AppDisplayName() =>
+            new Option<string>(
+                aliases: new[] { "--app-display-name" },
+                description: "App display name for Azure AD/AD B2C app registration creation.")
+            {
+                IsRequired = false
+            };
+
+        private static Option ProjectType() =>
+            new Option<string>(
+                aliases: new[] { "--project-type" },
```

Comment:
Do you give-up on the tool infering the project type.?

Reply:
Nope still using it if a project-type isn't passed from VS. Will be relying on it fully for the CLI experience as well. I will be adding stuff to make it more robust
- detecting multiple capabilities
- default project changes - what if they change Startup.cs to some other name (perfectly common scenario)
- change appsettings location

The current project type detection relies on the app to be pretty close a new app structure. However, I'd wager a good chunk of projects divert from that structure and would get erroneous behaviour. I don't have a timeline on these changes however.